src/main/java/org/apache/commons/compress/archivers/sevenz/LZMA2Decoder.java

```diff
@@ -33,9 +34,12 @@
 
     @Override
     InputStream decode(final String archiveName, final InputStream in, final long uncompressedLength,
-            final Coder coder, final byte[] password) throws IOException {
+            final Coder coder, final byte[] password, int maxMemoryLimitInKb) throws IOException {
         try {
             final int dictionarySize = getDictionarySize(coder);
+            if ((dictionarySize / 1024) > maxMemoryLimitInKb) {
```

Comment:
sorry about the previous comment which applied to `LZMAInputStream`. `LZMA2InputStream` also provides a static `getMemoryUsage` which might be a better option to calculate memory consumption.

Reply:
Code changed according to your comment.